src/proof.rs

```diff
@@ -22,166 +22,357 @@ use poseidon252::{
+            bid_stealth_addr,
+            bid_hashed_secret.var,
+            bid_elegibility_ts.var,
+            bid_expiration_ts.var,
+        );
+
+        // Add PI constraint for bid preimage check.
+        pi.push(PublicInput::BlsScalar(
+            -bid_hash.scalar,
+            composer.circuit_size(),
+        ));
+        // Constraint the hash to be equal to the real one
+        composer.constrain_to_constant(
+            computed_bid_hash,
+            BlsScalar::zero(),
+            -bid_hash.scalar,
+        );
+
+        // 3. t_a >= k_t
+        let third_cond = max_bound(
+            composer,
+            latest_consensus_round.scalar,
+            bid_elegibility_ts,
+        )
+        .0;
+        // We should get a 0 if t_e is greater, but we need this to be one in
+        // order to hold. Therefore we conditionally select one.
+        let third_cond = conditionally_select_one(composer, zero, third_cond);
+        // Constraint third condition to be true.
+        // So basically, that the rangeproofs hold.
+        composer.constrain_to_constant(
+            third_cond,
+            BlsScalar::one(),
+            BlsScalar::zero(),
+        );
+
+        // 4. t_e >= k_t
+        let fourth_cond = max_bound(
+            composer,
             latest_consensus_round.scalar,
-            latest_consensus_step.scalar,
-        ),
-    );
-    // 9. Score generation circuit check with the corresponding gadget.
-    prove_correct_score_gadget(
-        composer,
-        score,
-        bid_value,
-        secret_k,
-        bid_tree_root,
-        seed,
-        latest_consensus_round,
-        latest_consensus_step,
-    )?;
-    Ok(())
+            bid_expiration_ts,
+        )
+        .0;
+        // We should get a 0 if t_e is greater, but we need this to be one in
+        // order to hold. Therefore we conditionally select one.
+        let fourth_cond = conditionally_select_one(composer, zero, fourth_cond);
+        // Constraint fourth condition to be true.
+        // So basically, that the rangeproofs hold.
+        composer.constrain_to_constant(
+            fourth_cond,
+            BlsScalar::one(),
+            BlsScalar::zero(),
+        );
+
+        // 5. c = C(v, b) Pedersen Commitment check
+        let p1 = scalar_mul(composer, bid_value.var, GENERATOR_EXTENDED);
+        let p2 = scalar_mul(composer, bid_blinder.var, GENERATOR_NUMS_EXTENDED);
+        let computed_c = p1.point().fast_add(composer, *p2.point());
+        // Add PI constraint for the commitment computation check.
+        pi.push(PublicInput::AffinePoint(
+            bid.c,
+            composer.circuit_size(),
+            composer.circuit_size() + 1,
+        ));
+        // Assert computed_commitment == announced commitment.
+        composer.assert_equal_public_point(computed_c, bid.c);
+
+        // 6. 0 < value <= 2^64 range check
+        // v < 2^64
+        composer.range_gate(bid_value.var, 64usize);
```

Comment:
Isn't the range function naturally taking in a usize? If so, is the input of it here redundant?